File: matiasmasso/Mat.NET/DTO/DTOBreadCrumb.cs
using System.Collections.Generic;

namespace DTO
{
    public class DTOBreadCrumb
    {
        public List<BreadCrumbItem> Items { get; set; }

        public DTOBreadCrumb(DTOLang oLang, string sHomeEsp = "Inicio", string sHomeCat = "Inici", string sHomeEng = "Home", string sHomePor = "") : base()
        {
            Items = new List<BreadCrumbItem>();
            AddItem(oLang.Tradueix(sHomeEsp, sHomeCat, sHomeEng, sHomePor), "/");
        }

        public void AddItem(string sTitle, string sUrl = "")
        {
            BreadCrumbItem oItem = new BreadCrumbItem(sTitle, sUrl);
            Items.Add(oItem);
        }

        public static DTOBreadCrumb FromIncidencia(DTOLang oLang)
        {
            DTOBreadCrumb retval = new DTOBreadCrumb(oLang, "Formularios", "Formularis", "Forms");
            retval.AddItem(oLang.Tradueix("Incidencias", "Incidències", "Incidences"), "/incidencias");
            retval.AddItem(oLang.Tradueix("Incidencia", "Incidència", "Incidence"));
            return retval;
        }

        public static DTOBreadCrumb FromCategoriaDeNoticia(DTOCategoriaDeNoticia oCategoria)
        {
            DTOBreadCrumb retval = new DTOBreadCrumb(DTOApp.Current.Lang);
            retval.AddItem("Noticias", "/News");
            retval.AddItem(oCategoria.Nom);
            return retval;
        }

        public static DTOBreadCrumb FromNoticia(DTOLang oLang)
        {
            DTOBreadCrumb retval = new DTOBreadCrumb(oLang, "Inicio", "Inici", "Home");
            retval.AddItem(oLang.Tradueix("Noticias", "Notícies", "News"), "/News");
            return retval;
        }

        public static DTOBreadCrumb FromEvento(DTOLang oLang)
        {
            DTOBreadCrumb retval = new DTOBreadCrumb(oLang, "Inicio", "Inici", "Home");
            retval.AddItem(oLang.Tradueix("Eventos", "Esdeveniments", "Events"), "/Eventos");
            return retval;
        }
    }

    public class BreadCrumbItem
    {
        public string Title { get; set; }
        public string Url { get; set; }

        public BreadCrumbItem(string sTitle, string sUrl = "") : base()
        {
            Title = sTitle;
            Url = sUrl;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOBreadCrumb.vb
Public Class DTOBreadCrumb
    Public Property Items As List(Of BreadCrumbItem)

    Public Sub New(oLang As DTOLang, Optional sHomeEsp As String = "Inicio", Optional sHomeCat As String = "Inici", Optional sHomeEng As String = "Home", Optional sHomePor As String = "")
        MyBase.New()
        _Items = New List(Of BreadCrumbItem)
        AddItem(oLang.Tradueix(sHomeEsp, sHomeCat, sHomeEng, sHomePor), "/")
    End Sub

    Public Sub AddItem(sTitle As String, Optional sUrl As String = "")
        Dim oItem As New BreadCrumbItem(sTitle, sUrl)
        _Items.Add(oItem)
    End Sub

    Shared Function FromIncidencia(oLang As DTOLang) As DTOBreadCrumb
        Dim retval As New DTOBreadCrumb(oLang, "Formularios", "Formularis", "Forms")
        retval.AddItem(oLang.Tradueix("Incidencias", "Incidències", "Incidences"), "/incidencias")
        retval.AddItem(oLang.Tradueix("Incidencia", "Incidència", "Incidence"))
        Return retval
    End Function

    Shared Function FromCategoriaDeNoticia(oCategoria As DTOCategoriaDeNoticia) As DTOBreadCrumb
        Dim retval As New DTOBreadCrumb(DTOApp.current.lang)
        retval.AddItem("Noticias", "/News")
        retval.AddItem(oCategoria.Nom)
        Return retval
    End Function

    Shared Function FromNoticia(oLang As DTOLang) As DTOBreadCrumb
        Dim retval As New DTOBreadCrumb(oLang, "Inicio", "Inici", "Home")
        retval.AddItem(oLang.Tradueix("Noticias", "Notícies", "News"), "/News")
        Return retval
    End Function

    Shared Function FromEvento(oLang As DTOLang) As DTOBreadCrumb
        Dim retval As New DTOBreadCrumb(oLang, "Inicio", "Inici", "Home")
        retval.AddItem(oLang.Tradueix("Eventos", "Esdeveniments", "Events"), "/Eventos")
        Return retval
    End Function
End Class

Public Class BreadCrumbItem
    Public Property Title As String
    Public Property Url As String

    Public Sub New(sTitle As String, Optional sUrl As String = "")
        MyBase.New()
        _Title = sTitle
        _Url = sUrl
    End Sub
End Class


